Library/Homebrew/test/language/python_spec.rb

```diff
@@ -0,0 +1,124 @@
+require "language/python"
+require "resource"
+
+describe Language::Python::Virtualenv::Virtualenv do
+  subject { described_class.new(formula, dir, "python") }
+
+  let(:dir) { Pathname.new(Dir.mktmpdir) }
+
+  let(:resource) { double("resource", stage: true) }
+  let(:formula_bin) { dir/"formula_bin" }
+  let(:formula) { double("formula", resource: resource, bin: formula_bin) }
+
+  after(:each) do
+    dir.rmtree
+  end
+
+  describe "#create" do
+    it "creates a virtual environment" do
+      expect(formula).to receive(:resource).with("homebrew-virtualenv").and_return(resource)
+      subject.create
+    end
+
+    specify "virtual environment creation is idempotent" do
+      expect(formula).to receive(:resource).with("homebrew-virtualenv").and_return(resource)
+      subject.create
+      FileUtils.mkdir_p dir/"bin"
+      FileUtils.touch dir/"bin/python"
+      subject.create
+      FileUtils.rm dir/"bin/python"
+    end
+  end
+
+  describe "#pip_install" do
+    it "accepts a string" do
+      expect(formula).to receive(:system) do |*args|
+        expect(args.first).to eq(dir/"bin/pip")
+        expect(args).to include("install")
+        expect(args.last).to eq("foo")
```

Comment:
Could we use `receive(…).with(…).and_return(…)` here? I think block syntax for stubs should be a last resort where RSpec doesn't have built-in methods to cover our use case.